File: go-client-sdk/pkg/models/operations/getbatchcrmdeal.go
package operations

import (
	"github.com/speakeasy-sdks/orb-sdks/go-client-sdk/pkg/models/shared"
)

type GetBatchCrmDealQueryParams struct {
	AccessToken string  `queryParam:"style=form,explode=true,name=accessToken"`
	AllFields   *bool   `queryParam:"style=form,explode=true,name=allFields"`
	Ids         *string `queryParam:"style=form,explode=true,name=ids"`
}

type GetBatchCrmDealResponseBody struct {
	Deals      []shared.Deal `json:"deals,omitempty"`
	InvalidIds []string      `json:"invalidIds,omitempty"`
}

type GetBatchCrmDealRequest struct {
	QueryParams GetBatchCrmDealQueryParams
}

type GetBatchCrmDealResponse struct {
	ContentType  string
	StatusCode   int64
	ResponseBody *GetBatchCrmDealResponseBody
}

File: typescript-client-sdk/src/sdk/models/operations/getbatchcrmdeal.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class GetBatchCrmDealQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=accessToken" })
  accessToken: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=allFields" })
  allFields?: boolean;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=ids" })
  ids?: string;
}


export class GetBatchCrmDealResponseBody extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=deals", elemType: shared.Deal })
  deals?: shared.Deal[];

  @SpeakeasyMetadata({ data: "json, name=invalidIds" })
  invalidIds?: string[];
}


export class GetBatchCrmDealRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  queryParams: GetBatchCrmDealQueryParams;
}


export class GetBatchCrmDealResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  responseBody?: GetBatchCrmDealResponseBody;
}
